parser/parsing.go

```diff
@@ -376,42 +508,12 @@ func parseTypeTerm(toks []string) (Type, error) {
 		}
 
 		return &TypeList{
+			Kind:          "list",
```

Comment:
it seems like TypeStruct always has Kind "struct", and TypeList always has kind "list". maybe we can just encode that into the type somehow?